src/api_server.cc

```diff
@@ -44,21 +44,21 @@ void MetadataApiServer::Handler::operator()(const HttpServer::request& request,
     std::string id = request.destination.substr(kPrefix.size());
     std::lock_guard<std::mutex> lock(store_.resource_mu_);
     const auto result = store_.resource_map_.find(id);
-    if (result == store_.resource_map_.end()) {
+    if (result != store_.resource_map_.end()) {
```

Comment:
The overall change looks good. 

Not sure about this commit though - why the need to reverse the if/else?
Positive comparisons are easier to read (harder to misread) anyway.

Reply:
This was just to make the change from `if`/`else` to `try`/`catch` easier to see (otherwise it might get lost in the delete/add of the move).

Reply:
thanks. I was reading the commits the wrong way around.